### PR TITLE
Reusable cards

### PR DESCRIPTION
**Pull request by emmanuellosch**
This PR ist about my User Form. You can upload an Image, your name and write a status like: "Zero Waste Week 33" its the background for building a user profil card and for building a community between the app users. 
Also i changed the "Like-Button" to a heart - some basic styling to get further with the overall style. 

But i like to fetch all the data and get the image to the server - there is something missing yet ...

### CODE REVIEW

**Review by mrana17 — APPROVED** [review]
lgtm 🚀

File: client/src/App.js [hunk 1]
```diff
@@ -7,11 +7,26 @@ import Profil from "./pages/Profil";
...
+  });
+
+  const updateUploadedFiles = (files) =>
+    setNewUserInfo({ ...newUserInfo, profileImages: files });
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+  };
+  */
+
```
Comment (miriamhaenle) [inline client/src/App.js:29]: No commented out code in main! Remove commented out code before merging!

File: client/src/App.js [hunk 2]
```diff
@@ -7,11 +7,26 @@ import Profil from "./pages/Profil";
 import Navigation from "./components/Navigation";
 import saveToLocal from "./lib/saveToLocal";
 import loadFromLocal from "./lib/loadFromLocal";
+//import FileUpload from "./components/UploadComponent";
```
Comment (miriamhaenle) [inline client/src/App.js:10]: No commented out code on main ;)

File: client/src/GlobalStyles.js [hunk 3]
```diff
@@ -67,33 +67,55 @@ export const GlobalStyles = createGlobalStyle`
   max-width: 500px;
   margin: 50px auto;
   padding: 40px 25px;
-  background-color: whitesmoke;
-  font-family: Arial;
-  font-size: 16px;
+  //background-color: whitesmoke;
+  //font-family: Arial;
+  //font-size: 16px;
```
Comment (miriamhaenle) [inline client/src/GlobalStyles.js:72]: If you don't need the styles anymore delete it! Commenting out code and pushing it into main just clutters up your codebase. If you want to bring it back you can always check your git history

File: client/src/components/RecipeCard.js [hunk 4]
```diff
@@ -19,6 +20,8 @@ export default function RecipeCard({ addRecipeToFavorites, recipe }) {
     </Div>
```
Comment (miriamhaenle) [inline client/src/components/RecipeCard.js:20]: Maybe call the variable `Details` instead of `Text`?

File: client/src/components/RecipeCard.js [hunk 5]
```diff
@@ -167,4 +175,32 @@ const Card = styled.div`
...
+
+  background-color: #bebebe;
+  border: 2px solid #fff;
+  border-radius: 20px;
+  color: #922c88;
+  font-weight: bold;
+  padding: 1.2rem;
+  opacity: 0.7;
+  box-shadow: 0 0 5px rgba(75, 75, 75, 0.07);
+  text-transform: uppercase;
```
Comment (miriamhaenle) [inline client/src/components/RecipeCard.js:205]: Why do you use the additonal class `.btn` here inside your styled component? 
This is really confusing. Consider to either have two different styled button components or use props to dynamically change the designs that should defer.

File: client/src/components/UploadComponent.js [hunk 6]
```diff
@@ -0,0 +1,339 @@
+import React, { useState, useRef } from "react";
```
Comment (miriamhaenle) [inline client/src/components/UploadComponent.js:1]: No need to import React
```suggestion
import { useState, useRef } from "react";
```

File: client/src/components/UploadComponent.js [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+
+  const addNewFiles = (newFiles) => {
+    for (let file of newFiles) {
+      if (file.size < maxFileSizeInBytes) {
+        if (!otherProps.multiple) {
+          return { file };
+        }
+        files[file.name] = file;
+      }
+    }
```
Comment (miriamhaenle) [inline client/src/components/UploadComponent.js:36]: Does this actually work or maybe throw an error in the console? Usually the rule of hooks suggests not to use hooks inside a javascript function but only from a React functional component

File: client/src/pages/Bathroom.js [hunk 8]
```diff
@@ -1,31 +1,234 @@
 import { NavLink } from "react-router-dom";
+import React, { useState } from "react";
 
-export default function Bathroom() {
-  return (
-    <div className="bathroom-wrapper">
-      <h2>Bathroom</h2> <br />
+import styled from "styled-components";
+
+export default function Kitchen() {
```
Comment (miriamhaenle) [inline client/src/pages/Bathroom.js:6]: The filename says 'Bathroom' but the function says 'Kitchen' ?

File: client/src/pages/Bedroom.js [hunk 9]
```diff
@@ -0,0 +1,234 @@
+import { NavLink } from "react-router-dom";
+import React, { useState } from "react";
+
+import styled from "styled-components";
+
+export default function Kitchen() {
```
Comment (miriamhaenle) [inline client/src/pages/Bedroom.js:6]: The filename says 'Bedroom' but the function is called 'Kitchen'?!

File: client/src/components/UploadComponent.js [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+
+  const removeFile = (fileName) => {
+    delete files[fileName];
+    setFiles({ ...files });
+    callUpdateFilesCb({ ...files });
+  };
+
+  return (
+    <>
+      <FileUploadContainer>
```
Comment (miriamhaenle) [inline client/src/components/UploadComponent.js:62]: Remove the React fragments as the <FileUploadContainer> already wraps the whole content

File: client/src/pages/Bathroom.js [hunk 11]
```diff
@@ -1,31 +1,234 @@
...
+  const [showText, setShowText] = useState(false);
+  const onClick = () => setShowText(!showText);
+  const Text = () => (
```
Comment (miriamhaenle) [inline client/src/pages/Bathroom.js:9]: Details?

File: client/src/pages/Bathroom.js [hunk 12]
```diff
@@ -1,31 +1,234 @@
...
+      </Card>
+
+      <P className="return_home">
+        <NavLink to="/Reusable">&lt; Return to Reusable</NavLink>
+      </P>
     </div>
   );
 }
+
+const Card = styled.div`
```
Comment (miriamhaenle) [inline client/src/pages/Bathroom.js:87]: Your are mixing styled components and classNames. This sort of defeats the purpose/advantage of styled components

File: client/src/pages/Kidsroom.js [hunk 13]
```diff
@@ -1,31 +1,234 @@
 import { NavLink } from "react-router-dom";
+import React, { useState } from "react";
 
-export default function Kidsroom() {
-  return (
-    <div className="kidsroom-wrapper">
-      <h2>Kidsroom</h2> <br />
+import styled from "styled-components";
+
+export default function Kitchen() {
```
Comment (miriamhaenle) [inline client/src/pages/Kidsroom.js:6]: Kidsroom vs. Kitchen

File: client/src/pages/Livingroom.js [hunk 14]
```diff
@@ -1,31 +1,234 @@
 import { NavLink } from "react-router-dom";
+import React, { useState } from "react";
```
Comment (miriamhaenle) [inline client/src/pages/Livingroom.js:2]:
```suggestion
import { useState } from "react";
```

File: client/src/pages/Profil.js [hunk 15]
```diff
@@ -1,9 +1,114 @@
...
+        status: "",
+      };
+    }
+    return {
+      ...state,
+      [event.name]: event.value,
+    };
+  };
+
+  const [formData, setFormData] = useReducer(formReducer, {});
```
Comment (miriamhaenle) [inline client/src/pages/Profil.js:24]: Why did you choose to use the useReducer hook for the form? As it is a rather advanced concept im curious :)

File: client/src/pages/Profil.js [hunk 16]
```diff
@@ -1,9 +1,114 @@
...
+
+  const updateUploadedFiles = (files) =>
+    setNewUserInfo({ ...newUserInfo, profileImages: files });
+
+  const handleSubmit = (event) => {
```
Comment (miriamhaenle) [inline client/src/pages/Profil.js:29]: I've noticed that you called the `event` only `e` in another file. Be consistent when using this. So either always write `event` or always use `e`

File: client/src/pages/Profil.js [hunk 17]
```diff
@@ -1,9 +1,114 @@
...
+    event.preventDefault();
+    event.preventDefault();
```
Comment (miriamhaenle) [inline client/src/pages/Profil.js:31]: I think once is enough :)

**Review by miriamhaenle — CHANGES_REQUESTED** [review]
Hey Emmanuel, I added a few comments in the code.
Please take a look, some of my comments also apply to other documents so I only added it once :)

**Pull request closed without merging**